### PR TITLE
:sparkles:Bump to golangci-lint v1.45.2

### LINKED ISSUE #6163: Bump to golangci-lint v1.44.1
golangci-lint v1.44.1 was just released, let's use it :)

Tasks:
* [ ] bump golangci-lint in .github/workflows/golangci-lint.yml
* [ ] run make lint and potentially fix findings
* [ ] the new release contains thelper v0.5.1 which fixes a panic. For that panic we had to introduce a workaround in https://github.com/kubernetes-sigs/cluster-api/pull/6014/files that we now should be able to revert.
  * (just grep for `// TODO(sbueringer) We are storing the func in testFunc temporarily to work around`)

/kind feature
/good-first-issue

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

**What this PR does / why we need it**:
- Bump to golangci-lint v1.44.2
- Fix lint findings
- Revert panic workaround

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6163

### CODE REVIEW

**Comment by sbueringer** [comment]
Thx!

/lgtm

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -17,5 +17,5 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v2
         with:
-          version: v1.44.0
+          version: v1.45.0
```
Comment (sbueringer) [inline .github/workflows/golangci-lint.yml:20]: Just saw it, can you please bump to v1.45.2? (was just released yesterday)

Thank you very much!

(there are some go 1.8 related fixes in v1.45.1)

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Comment by sbueringer** [comment]
@Prajyot-Parab Thank you very much!!

/lgtm
/approve

**Pull request merged**